tests/e2e/lib/flows/jetpack-connect.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+/**
+ * Internal dependencies
+ */
+import DashboardPage from '../pages/wp-admin/dashboard';
+import Sidebar from '../pages/wp-admin/sidebar';
+import JetpackPage from '../pages/wp-admin/jetpack';
+import LoginPage from '../pages/wpcom/login';
+import AuthorizePage from '../pages/wpcom/authorize';
+import JetpackSiteTypePage from '../pages/jetpack-connect/site-type';
+import JetpackSiteTopicPage from '../pages/jetpack-connect/site-topic';
+import JetpackUserTypePage from '../pages/jetpack-connect/user-type';
+import PickAPlanPage from '../pages/wpcom/pick-a-plan';
+import HomePage from '../pages/wpcom/home';
+import WPLoginPage from '../pages/wp-admin/login';
+import CheckoutPage from '../pages/wpcom/checkout';
+import ThankYouPage from '../pages/wpcom/thank-you';
+import MyPlanPage from '../pages/wpcom/my-plan';
+
+const cookie = config.get( 'storeSandboxCookieValue' );
+const cardCredentials = config.get( 'testCardCredentials' );
+
+/**
+ * Connects your site to WPCOM as `wpcomUser`, buys a Professional plan via sandbox cookie
+ * @param {Object} o Optional object with params such as `wpcomUser` and expected Jetpack plan
+ */
+export async function connectThroughWPAdminIfNeeded( {
+	wpcomUser = 'defaultUser',
+	plan = 'pro',
+} = {} ) {
+	await ( await HomePage.visit( page ) ).setSandboxModeForPayments( cookie );
+	await ( await WPLoginPage.visit( page ) ).login();
+	await ( await DashboardPage.init( page ) ).setSandboxModeForPayments( cookie, '.eu.ngrok.io' );
+	await ( await Sidebar.init( page ) ).selectJetpack();
+
+	const jetpackPage = await JetpackPage.init( page );
+	if ( await jetpackPage.isConnected() ) {
+		await jetpackPage.openMyPlan();
+		if ( await jetpackPage.isPlan( plan ) ) {
+			// eslint-disable-next-line no-console
```

Comment:
If there are likely going to be lots of these in future, you could also add a new override here:

https://github.com/Automattic/jetpack/blob/91939c13e6d278629deb444ff3538677deb009d8/.eslintrc.js#L102

```js
files: [ ' tests/e2e/**/*' ],
rules: {
	'no-console': 0
}
```

Reply:
Thanks! I'd like to switch to some fancier logging solution, so for now, it's fine :)

Reply:
Sounds great! :-)